### PR TITLE
Test/parking detection

### PR DESCRIPTION
**Pull request by mikeslgoh**
- Refactored ParkingDetection script to have a Behaviour class and the main bounds calculation separated from that so that it can be tested.
- Tested with 4 test cases:
1. car bounds inside parking bounds perfectly
2. car bounds is rotated to fit into parking bounds
3. car bounds is rotated to be outside parking bounds
4. car bounds is outside parking bounds

### CODE REVIEW

**Comment by mikeslgoh** [comment]
[Refactor ParkingDetection for testing](https://trello.com/c/u27KXpWU/313-refactor-parkingdetection-for-testing)

File: Assets/_PartyParkade/Tests/ParkingDetection_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+        
+        [Test]
+        public void InsideParkingSpot_CalculateBounds()
+        {
+            Vector3 centerOfCarBounds = Vector3.zero;
+            Bounds carBounds = new Bounds(centerOfCarBounds, new Vector3(2, 1, 1));
+            Bounds parkingSpotBounds = new Bounds(Vector3.zero, new Vector3(2, 1, 1));
+            
+            Assert.IsTrue(parkingDetection.CalculateBounds(centerOfCarBounds,
```
Comment (m4rtint) [inline Assets/_PartyParkade/Tests/ParkingDetection_Tests.cs:35]: parkingDetection.CalculateBounds

move this method into it's own variable, so it's more readible
so it's like 
bool actualResult = parkingDetectiong.Calculate BOunds

File: Assets/_PartyParkade/Tests/ParkingDetection_Tests.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            parkingDetection = null;
+        }
+
+        
+        [Test]
+        public void InsideParkingSpot_CalculateBounds()
+        {
+            Vector3 centerOfCarBounds = Vector3.zero;
+            Bounds carBounds = new Bounds(centerOfCarBounds, new Vector3(2, 1, 1));
+            Bounds parkingSpotBounds = new Bounds(Vector3.zero, new Vector3(2, 1, 1));
```
Comment (m4rtint) [inline Assets/_PartyParkade/Tests/ParkingDetection_Tests.cs:33]: If you could separate the variables 
new Vector3(2, 1, 1), into their own variables, so others know what they represent, that'll be great.

File: Assets/_PartyParkade/Script/Game/ParkingDetection.cs [hunk 3]
```diff
@@ -1,53 +1,33 @@
-﻿//-----------------------------------------------------------------------
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/Game/ParkingDetection.cs:1]: Same complaint, should add this back

File: Assets/_PartyParkade/Tests/ParkingDetection_Tests.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
+﻿using System.Collections;
```
Comment (Gursaanj) [inline Assets/_PartyParkade/Tests/ParkingDetection_Tests.cs:1]: Missing Header

**Pull request merged**